src/finance_quote/modules/fixerio.py

```diff
@@ -0,0 +1,185 @@
+"""
+Fixer.io currency rates 
+Currently only implements the free public API access with yesterday's rates.
+"""
+import os
+import tempfile
+from datetime import datetime
+import logging
+from decimal import Decimal
+
+from finance_quote.base import Source, Quote
+import finance_quote
+
+try: import simplejson as json
+except ImportError: import json
+
+
+class FixerioQuote(Quote):
+    """ Fixer.io-specific quote """
+    pass
+
+
+class Fixerio(Source):
+    """Retrieves prices from data files or online provider(s)"""
+    def __init__(self):
+        self.cache_path = tempfile.gettempdir()
+        self.logger = logging.getLogger(__name__)
+
+    def download(self, namespace: str, mnemonic: str, currency: str) -> FixerioQuote:
+        """ Download latest rates. Caches into temp directory. """
```

Comment:
can the methods in Source be implemented instead of the download method ?

Reply:
Yes but I would prefer if we discuss the design first before we try to implement it. What is the purpose of the Source class?

Reply:
To define a common interface any source can/must implement. 
Then, other parts of the lib (CLI, caching , etc) know how they can interact with a source

Reply:
OK. That's what I thought was implicit. Again, following the f::q interface, there *is* only one method - download. Everything else, including caching, is implementation-specific.
Of course, we can change that, but that is/was not the original goal.

Reply:
and what is "downloaded" ? the full history ? today's closing ? live prices if call during the day? and how you specify that you only what a specific day or interval ?
I am not familiar with f::q ...

Reply:
> I am not familiar with f::q
Then that's the first thing I would suggest. :)